src/main/java/org/broadinstitute/dsm/jobs/PubSubLookUp.java

```diff
@@ -0,0 +1,144 @@
+package org.broadinstitute.dsm.jobs;
+
+import com.google.gson.Gson;
+import com.google.pubsub.v1.PubsubMessage;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.ddp.db.SimpleResult;
+import org.broadinstitute.dsm.model.KitDDPNotification;
+import org.broadinstitute.dsm.model.birch.DSMTestResult;
+import org.broadinstitute.dsm.model.birch.TestBostonResult;
+import org.broadinstitute.dsm.util.EventUtil;
+import org.json.JSONString;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.ResultSet;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Arrays;
+
+import static org.broadinstitute.ddp.db.TransactionWrapper.inTransaction;
+
+public class PubSubLookUp {
+    private static final Logger logger = LoggerFactory.getLogger(PubSubLookUp.class);
+    private static String SELECT_LATEST_RESULT_QUERY = "SELECT kit.test_result FROM dev_dsm_db.ddp_kit_request req LEFT JOIN  ddp_kit kit ON (kit.dsm_kit_request_id = req.dsm_kit_request_id) " +
+            "WHERE req.external_order_number = ? ";
+
+    public static void processCovidTestResults(PubsubMessage message) {
+        String data = message.getData().toStringUtf8();
+        TestBostonResult testBostonResult = new Gson().fromJson(data, TestBostonResult.class);
+        if (shouldWriteResultIntoDB(testBostonResult)) {
+            writeResultsIntoDB(testBostonResult);
+            tellPepperAboutTheNewResults(testBostonResult);// notify pepper if we update DB
+        }
+    }
+
+    private static boolean shouldWriteResultIntoDB(TestBostonResult testBostonResult) {
+        DSMTestResult[] dsmTestResultArray = getLatestKitTestResults(testBostonResult);
+        if (dsmTestResultArray == null) {
+            return true;
+        }
+        DSMTestResult dsmTestResult = null;
+        if (dsmTestResultArray != null && dsmTestResultArray.length > 0) {
+            dsmTestResult = dsmTestResultArray[dsmTestResultArray.length - 1];
+        }
+
+        //corrected result -> assuming corrected results are always changed
+        if (dsmTestResult != null && !testBostonResult.isCorrected() && dsmTestResult.isCorrected) {
+            return true;
+        }
+        //duplicate result
+        if (testBostonResult.isCorrected() == dsmTestResult.isCorrected
+                && StringUtils.isNotBlank(dsmTestResult.result) && dsmTestResult.result.equals(testBostonResult.getResult())
+                && StringUtils.isNotBlank(dsmTestResult.date) && dsmTestResult.date.equals(testBostonResult.getTimeCompleted())) {
+            return false;
+        }
+        // weird result
+        if (!testBostonResult.isCorrected()
+                && StringUtils.isNotBlank(dsmTestResult.result) && !dsmTestResult.result.equals(testBostonResult.getResult()) ) {
```

Comment:
Should this just be `new result not corrected`? So that if new result is not corrected and we have existing result, we error?

Reply:
do we really want that as an error. maybe warning would be enough

Reply:
@yufengwng I was thinking if we get the same result as corrected and as not, do we care?

@SimoneMaiwald Wouldn't we wanna know if there are dual results for a participant?

Reply:
@pegahtah as long as we don't save it and don't send it over to pepper, then it's fine with me.

@SimoneMaiwald yeah a `warn()` should be good, as long as we throw it out.